mirgecom/filter.py

```diff
@@ -44,3 +45,51 @@ def get_spectral_filter(dim, order, cutoff, filter_order):
         filter[m, m] = np.exp(-1.0 * alpha
                               * ((m - nstart) / nfilt) ** filter_order)
     return filter
+
+
+class SpectralFilter:
```

Comment:
I think this can be done as just a function, by memoizing the filter matrix in the discretization and the kernel in the array context, using `@memoize_in`. This relieves the user from having to worry about the lifetime of this `SpectralFilter` object.

Reply:
If I am following, then that means embedding the filter data into a grudge data structure, and the kernel code into a kernel interface in meshmode?